### ISSUE TITLE
DoReFa validation accuracy increases with the bitwidth reduction of activate and weight values

### ISSUE THREAD

**Issue by tunghoang290780** [tensorpack/tensorpack#1305]
Hi All,

I modify DoReFa to use VGG11 with Cifar10. I evaluate three cases W,A,G=8,8,32, W,A,G=4,4,32, W,A,G=2,2,32 with 50 epochs (BatchNorm is used for large batch size=256). The unexpected problem is the validation accuracy of W,A,G=8,8,32 is less than W,A,G=4,4,32 which, in my opinion, could be incorrect. Meanwhile, the validation accuracy for W,A,G=2,2,32 is lowest which makes sense.

Could anybody give hints to explain why the validation of W,A,G=8,8,32 could be less than W,A,G=4,4,32 ?

Thanks
/TH

**Comment by ppwwyyxx** [comment]
That's why Cifar10 should not be used to test model compression algorithm..

Also, you should expect training accuracy to be lower for 8,8,32, not validation accuracy.

**Comment by tunghoang290780** [comment]
Thanks,

Indeed, that is training accuracy (it's printed with "val_" which makes me confuse). Perhaps, I should move to ImageNet

**Comment by ppwwyyxx** [comment]
If it starts with `val_` then it's validation accuracy.
